### ISSUE TITLE
Some functions should probably be `const`

### ISSUE THREAD

**Issue by giannitedesco** [yoanlcq/vek#98]
Lots of methods could be const, particularly constructors, and conversions from column/row oriented and back.

That way you can have static/const vectors/matrices etc..

**Comment by yoanlcq** [comment]
There was a discussion on this in #12 , you may want to search for `The only function I've been able to make const was new() for vectors` in that thread.

Then again, that may be outdated. It's been a few years...

**Comment by giannitedesco** [comment]
Yeah pretty sure it is, I can't see why broadcast, conversions, maps, casts, the like can't all be const

**Comment by yoanlcq** [comment]
I've given it a shot today and unfortunately `broadcast` was the only one I could make `const`.

As you say, it's obvious that casts/conversions/constructors should and could be `const` when `T` is a primitive type, the problem is that I'm dealing with generic structs and traits, which is a context where `const` is hard to guarantee.

So even simple stuff such as `zero()` won't work because I'm relying on the `Zero` trait, and for all other functions, I'm getting `error[E0493]: destructor of `...` cannot be evaluated at compile-time`. Even on `into_tuple` and `into_array`.

I would suggest that you try making these changes to see for yourself, but honestly I think the return on investment is too low.

There could be significant improvements made if there was some way to specify a bound such as `T : destructor-is-const`, that would help get rid of E0493.

**Comment by giannitedesco** [comment]
Ah I see, makes sense, I didn't look closely enough there, I just assumed we're always dealing with floats. Fair enough.
